File: android/src/main/java/dragon3/stage/StageBack.java
package dragon3.stage;

import java.util.LinkedHashMap;
import java.util.Map;

import lombok.Getter;

public enum StageBack {
	WHITE("無地", "back_WHITE.png", "obj_BASIC.png"),
	GRASS("草原", "back_GRASS.png", "obj_BASIC.png"),
	SAND("砂地", "back_SAND.png", "obj_BASIC.png"),
	
	;
	
	@Getter private String text;
	@Getter private String backImage;
	@Getter private String objImage;


	private StageBack(String text, String backImage, String objImage) {
		this.text = text;
		this.backImage = backImage;
		this.objImage = objImage;
	}	
	
	
	public static Map<String, String> createMap() {
		Map<String, String> idAndText = new LinkedHashMap<>();
		for (StageBack a : values()) {	
			idAndText.put(a.name(), a.name());
		}
		return idAndText;
	}


}

File: core/src/main/java/dragon3/stage/StageBack.kt
package dragon3.stage

import java.util.LinkedHashMap


enum class StageBack private constructor(
        val text: String,
        val backImage: String,
        val objImage: String) {
    WHITE("無地", "back_WHITE.png", "obj_BASIC.png"),
    GRASS("草原", "back_GRASS.png", "obj_BASIC.png"),
    SAND("砂地", "back_SAND.png", "obj_BASIC.png");


    companion object {


        fun createMap(): Map<String, String> {
            val idAndText = LinkedHashMap<String, String>()
            for (a in values()) {
                idAndText.put(a.name, a.name)
            }
            return idAndText
        }
    }


}
